allennlp/models/coreference_resolution/coref.py

```diff
@@ -0,0 +1,244 @@
+        top_span_labels = util.batched_index_select(span_labels.unsqueeze(-1),
+                                                    top_span_idx,
+                                                    flat_top_span_idx) # [b, k, 1]
+
+        # Compute indices for antecedent spans to consider.
+        max_ant = min(self._max_antecedents, k) # []
+        target_idx = util.get_indices(k, is_cuda).unsqueeze(1) # [k, 1]
+        ant_offsets = (util.get_indices(max_ant, is_cuda) + 1).unsqueeze(0) # [1, max_ant]
+        raw_ant_idx = target_idx - ant_offsets # [k, max_ant]
+        ant_mask = (raw_ant_idx >= 0).float().unsqueeze(0).log() # [1, k, max_ant]
+        ant_idx = F.relu(raw_ant_idx.float()).long() # [k, max_ant]
+
+        # Select tensors relating to the antecedent spans.
+        ant_mention_scores = util.flattened_index_select(top_span_mention_scores,
+                                                         ant_idx).squeeze(-1) # [b, k, max_ant]
+        ant_emb = util.flattened_index_select(top_span_emb, ant_idx) # [b, k, max_ant, emb]
+        ant_labels = util.flattened_index_select(top_span_labels, ant_idx).squeeze(-1) # [b, k, max_ant]
+        ant_labels += ant_mask.long() # [b, k, max_ant]
+
+        # Expand top spans in preparation for pairwise comparisons.
+        target_mention_scores = top_span_mention_scores.expand(batch_size, k, max_ant) # [b, k, max_ant]
+        target_emb = top_span_emb.unsqueeze(2).expand(batch_size,
+                                                      k,
+                                                      max_ant,
+                                                      top_span_emb.size(-1)) # [b, k, max_ant, emb]
+        target_labels = top_span_labels.expand(batch_size, k, max_ant) # [b, k, max_ant]
+
+        # Compute antecedent scores.
+        similarity_emb = ant_emb * target_emb # [b, k, max_ant, emb]
+        ant_distance_emb = self._distance_embedding(self._bucket_distance(ant_offsets)) # [1, max_ant, emb]
+        ant_distance_emb = ant_distance_emb.unsqueeze(0) # [1, 1, max_ant, emb]
+        ant_distance_emb = ant_distance_emb.expand(batch_size,
+                                                   k,
+                                                   max_ant,
+                                                   ant_distance_emb.size(-1)) # [b, k, max_ant, emb]
+        pairwise_emb = torch.cat([target_emb,
+                                  ant_emb,
+                                  similarity_emb,
+                                  ant_distance_emb], -1) # [b, k, max_ant, emb]
+        ant_scores = self._antecedent_scorer(self._antecedent_feedforward(pairwise_emb)) # [b, k, max_ant, 1]
+        ant_scores = ant_scores.squeeze(-1) # [b, k, max_ant]
+        ant_scores += target_mention_scores + ant_mention_scores # [b, k, max_ant]
+        ant_scores += ant_mask # [b, k, max_ant]
```

Comment:
Calling this a mask confused me - we've typically used `mask` to refer to the binary tensor, not the log of the binary tensor.

Reply:
Fixed.